crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -609,13 +620,25 @@ impl SingleDiskPlot {
                 .map_mut(&metadata_file)?
         };
 
+        let plot_path = directory.join(Self::PLOT_FILE);
         let plot_file = OpenOptions::new()
             .read(true)
             .write(true)
             .create(true)
-            .open(directory.join(Self::PLOT_FILE))?;
+            .open(plot_path.clone())?;
+
+        let available_space = fs2::available_space(plot_path.clone())?;
+        let plot_size = PLOT_SECTOR_SIZE * target_sector_count;
```

Comment:
This doesn't account for metadata, some databases (which are tricky to compute and this is why it wasn't implemented yet) and the fact that plot might have been already pre-allocated.

Reply:
Thanks for the review!

I guess, that I can add metadata preallocation checks and pre-allocated `plot` checks.
But at this moment I'm not very familiar with the `parity-db` and `subspace` codebase, so I'm not sure about:
- How to make predictions about `log*` size. I've found the min/max sizes of one file, but I'm not sure about the number of possible log file counts.
- How to make predictions about `table*` size. I can guess, that number of rows can be limited by `provider_cache_size` and `piece_cache_size`, but I'm not sure about the final disk space, due to the unknowledge of the certain size of the row.

Reply:
The tricky part with databases is that they are not deterministic. Meaning they can actually grow over total expected amount of data we store in them and then do some cleanup (it is called differently in different databases) for various technical reasons.

I don't have answer as to how exactly ParityDB behaves and what guarantees we can have about it.

I think closing PR is fine, you can always reopen it or create another one instead if it has changed significantly.